packages/flutter/lib/src/widgets/platform_view.dart

```diff
@@ -581,6 +581,129 @@ class _UiKitPlatformView extends LeafRenderObjectWidget {
   }
 }
 
+/// The parameters used to create a [PlatformViewController].
+///
+/// See also [CreatePlatformView] which uses this object to create a [PlatformViewController].
+class PlatformViewCreationParams {
+
+  const PlatformViewCreationParams._({
+    @required this.id,
+    @required this.onPlatformViewCreated}):assert(id != null),
+                          assert(onPlatformViewCreated != null);
+
+  /// The auto generated id for the newly created platform view.
+  ///
+  /// [PlatformViewController.viewId] should match this id.
+  final int id;
+
+  /// Notifies when the PlatformView is ready.
+  ///
+  /// Must be invoked as soon as the embedded platform view is ready.
+  /// For example, on iOS, this has to be called when we know the `UIView` is initialized.
+  final PlatformViewCreatedCallback onPlatformViewCreated;
+}
+
+/// The factory to construct a [PlatformViewSurface].
+///
+/// It is required when constructing a [PlatformViewLink].
+typedef PlatformViewSurfaceFactory = PlatformViewSurface Function(BuildContext context, PlatformViewController controller);
+
+/// Constructs a [PlatformViewController].
+///
+/// The implementer of a new platform view is responsible to implement this method when constructing a [PlatformViewLink].
+typedef CreatePlatformView = PlatformViewController Function(PlatformViewCreationParams params);
+
+/// The widget responsible for handling the life cycle and focus for a platform view.
+///
+/// This widget provides simpler way to implement a new platform view.
+/// To implement a new platform view widget, return this widget in the `build` method.
+/// For example:
+/// ```dart
+/// class FooPlatformView extends StatelessWidget {
+///   @override
+///   Widget build(BuildContext context) {
+///     return PlatformViewLink(
+///       createCallback: createFooWebView,
+///       surfaceFactory: (BuildContext context, PlatformViewController controller, int id) {
+///        return PlatformViewSurface(
+///            gestureRecognizers: gestureRecognizers,
+///            controller: controller,
+///            hitTestBehavior: PlatformViewHitTestBehavior.opaque,
+///        );
+///       },
+///    );
+///   }
+/// }
+/// ```
+class PlatformViewLink extends StatefulWidget {
```

Comment:
Document what happens when the widget is rebuild with a different create callback or a different surfaceFactory

Reply:
done, with the change that made sure `PlatformViewSurface` only created once.